rotkehlchen/inquirer.py

```diff
@@ -920,6 +924,112 @@ def find_curve_pool_price(
+                return None
+        else:
+            lp_token = token.evm_address
+
+        lp_contract = EvmContract(
+            address=lp_token,
+            abi=node_inquirer.contracts.abi('GEARBOX_LP'),
+            deployed_block=0,  # not used here
+        )
+        if maybe_underlying_tokens is None or len(maybe_underlying_tokens) != 1:
+            # underlying token not recorded in the DB. Ask the chain
+            try:
+                remote_underlying_token = lp_contract.call(node_inquirer, 'underlyingToken')
+            except (RemoteError, BlockchainQueryError) as e:
+                log.error(f'Failed to query underlying token method in {node_inquirer.chain_name} Gearbox pool. {e!s}')  # noqa: E501
+                return None
+
+            try:
+                underlying_token_address = deserialize_evm_address(remote_underlying_token)
+            except DeserializationError:
+                log.error(f'underlying token call of {token.evm_address} returned invalid address {remote_underlying_token}')  # noqa: E501
+                return None
+
+            try:  # make sure it's in the global DB
+                underlying_token = get_or_create_evm_token(
+                    userdb=node_inquirer.database,
+                    evm_address=underlying_token_address,
+                    chain_id=token.chain_id,
+                    encounter=TokenEncounterInfo(
+                        description='Detecting Gearbox pool underlying tokens',
+                    ),
+                )
+            except NotERC20Conformant as e:
+                log.error(f'Error fetching {node_inquirer.chain_name} token {underlying_token_address} while detecting underlying tokens of {token.evm_address!s}: {e!s}')  # noqa: E501
+            # store it in the DB, so next time no need to query chain
+            with globaldb.conn.write_ctx() as write_cursor:
+                globaldb._add_underlying_tokens(
+                    write_cursor=write_cursor,
+                    parent_token_identifier=token.identifier,
+                    underlying_tokens=[
+                        UnderlyingToken(
+                            address=underlying_token_address,
+                            token_kind=EvmTokenKind.ERC20,
+                            weight=ONE,  # all gearbox vaults have single underlying
+                        ),
+                    ],
+                    chain_id=token.chain_id,
+                )
+        else:
+            underlying_token = EvmToken(
+                evm_address_to_identifier(
+                    address=maybe_underlying_tokens[0].address,
+                    chain_id=token.chain_id,
+                    token_type=EvmTokenKind.ERC20,
+                ),
+            )
+
+        if underlying_token is None:
+            log.error(f'Error fetching {node_inquirer.chain_name} token while detecting underlying tokens of {token.evm_address!s}')  # noqa: E501
+            return None
```

Comment:
how can it be none? The logic should have failed before if it wasn't a valid gearbox token before right?

Reply:
Was getting linting error because I missed the `return None` earlier, this is not needed anymore, removed.